File: Taejin/DS_Module/DequeExample.py
# DequeExample.py
import collections

dq = collections.deque()

# Push values
dq.appendleft( 2 ) # push front
dq.appendleft( 1 )
dq.append( 3 ) # push back

# Pop values
print( dq.popleft( ) ) # pop front
print( dq.pop( ) ) # pop back

# Front and Back values of deque
print( dq[0] )
print( dq[-1] )

File: Taejin/DS_Module/DequeExample.cpp
// DequeExample.cpp
#include <iostream>
#include <deque>

using namespace std;

int main(void) {
	// Declaration
	deque<int> dq;

	// Push values
	dq.push_front( 2 );
	dq.push_front( 1 );
	dq.push_back( 3 );

	// Pop values
	dq.pop_front( );
	dq.pop_back( );

	// Front and Back of deque
	cout << dq.front() << ' ' << dq.back() << '\n';

	return 0;
}